app/controllers/concerns/report_embeddable.rb

```diff
@@ -32,7 +33,10 @@ def build_report_data(options = {})
   # returns true if no errors, false otherwise
   def run_and_handle_errors
     begin
-      @report.run(current_ability)
+      Rails.cache.fetch(@report.cache_key_with_responses(current_mission)) do
+        @report.run(current_ability)
+        @report
+      end
```

Comment:
Oh, i misinterpreted what you were asking for. (I though the problem was on the block code). I'll be doing these changes (including the change to self) It will make everything clearer.

Thanks!